### PR TITLE
add informative error message regarding symbolic dimensions

### PR DESCRIPTION
**Pull request by jgbradley1**
**Description**: Adds a better error message when creating tensors

**Motivation and Context**
- Fixes https://github.com/microsoft/onnxruntime/issues/4296

### CODE REVIEW

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 1]
```diff
@@ -162,6 +162,9 @@ ORT_STATUS_PTR CreateTensorImpl(MLDataType ml_type, const int64_t* shape, size_t
     shapes[i] = shape[i];
   }
 
+  if ( std::any_of(shapes.begin(), shapes.end(), [](int i){return i<0;}) )
```
Comment (skottmckay) [inline onnxruntime/core/session/onnxruntime_c_api.cc:165]: nit: could do this check when populating the 'shapes' vector. would also be good to run clang-format so the formatting is consistent.

The build failure is due to shapes containing int64_t but your lambda has an 'int' for the argument.

Comment (jgbradley1) [reply onnxruntime/core/session/onnxruntime_c_api.cc]: Reformatted the code and moved the check inside the for loop. clang-format also corrected formatting in another part of the file as well.

**Comment by jgbradley1** [comment]
Am I able to kick off the CI jobs with a comment or is that for repo owners/maintainers only?

**Comment by snnn** [comment]
Our CI system has an on-going issue. Please retry it later.

**Pull request merged**